lib/resty/iprepd.lua

```diff
@@ -56,23 +56,28 @@ function _M.new(options)
     statsd_max_buffer_count =  options.statsd_max_buffer_count or 100,
     statsd_flush_timer = options.statsd_flush_timer or 5,
     dont_block = options.dont_block or 0,
+    verbose = options.verbose or 0,
     whitelist = whitelist,
   }
 
   return setmetatable(self, mt)
 end
 
 function _M.check(self, ip)
+  self:debug_log("Checking " .. ip)
   ngx.req.set_header('X-Foxsec-IP-Reputation-Below-Threshold', 'false')
   ngx.req.set_header('X-Foxsec-Block', 'false')
   if self.whitelist then
     if iputils.ip_in_cidrs(ip, self.whitelist) then
+      self:debug_log(ip .. " in whitelist")
       return
     end
   end
 
+
   local reputation = self:get_reputation(ip)
   if reputation then
+    self:debug_log("Got reputation of " .. reputation .. " for " .. ip)
```

Comment:
Using `string.format` is much faster than concat'ing strings if there are multiple elements to combine, but it's probably not a big deal for debug output.